### PR TITLE
`flutter create --template=plugin` now includes flutter (dart) unit test

### PR DESCRIPTION
**Pull request by truongsinh**

### CODE REVIEW

File: packages/flutter_tools/templates/plugin/pubspec.yaml.tmpl [hunk 1]
```diff
@@ -11,6 +11,11 @@ dependencies:
   flutter:
     sdk: flutter
 
+dev_dependencies:
+  mockito: 3.0.0
```
Comment (jonahwilliams) [inline packages/flutter_tools/templates/plugin/pubspec.yaml.tmpl]: 4.0.0

File: packages/flutter_tools/templates/plugin/test/projectName_test.dart.tmpl [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  {{pluginDartClass}} {{projectName}};
+
+  setUp(() {
+    methodChannel = MockMethodChannel();
+    {{projectName}} = {{pluginDartClass}}.private(methodChannel);
+  });
+
+  test('getPlatformVersion', () async {
+    when(methodChannel.invokeMethod('getPlatformVersion'))
+        .thenAnswer((Invocation invoke) => Future<String>.value('42'));
```
Comment (jonahwilliams) [inline packages/flutter_tools/templates/plugin/test/projectName_test.dart.tmpl]:
```suggestion
        .thenAnswer((Invocation invocation) async => '42');
```

File: packages/flutter_tools/templates/plugin/test/projectName_test.dart.tmpl [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    expect(await {{projectName}}.platformVersion, '42');
+  });
+}
+
+class MockMethodChannel extends Mock implements MethodChannel {}
```
Comment (jonahwilliams) [inline packages/flutter_tools/templates/plugin/test/projectName_test.dart.tmpl]:
```suggestion
class MockMethodChannel extends Mock implements MethodChannel {}

```

File: packages/flutter_tools/templates/plugin/lib/projectName.dart.tmpl [hunk 4]
```diff
@@ -1,12 +1,26 @@
 import 'dart:async';
 
 import 'package:flutter/services.dart';
+import 'package:meta/meta.dart' show visibleForTesting;
 
 class {{pluginDartClass}} {
-  static const MethodChannel _channel =
-      const MethodChannel('{{projectName}}');
+  factory {{pluginDartClass}}() {
```
Comment (jonahwilliams) [inline packages/flutter_tools/templates/plugin/lib/projectName.dart.tmpl]: I don't really think this is necessary to make the channel testable. There is already a method on `MethodChannel.setMockMethodCallHandler` for testing the dart side

Comment (truongsinh) [reply packages/flutter_tools/templates/plugin/lib/projectName.dart.tmpl]: thank you for your suggestion, I was following example from https://github.com/flutter/plugins/blob/master/packages/battery/lib/battery.dart#L14 . Let me do further research and  resolve this.

File: packages/flutter_tools/templates/plugin/test/projectName_test.dart.tmpl (hunk 3 above)
Comment (truongsinh) [reply packages/flutter_tools/templates/plugin/test/projectName_test.dart.tmpl]: is this neccessary? this file already has empty line at the end, if we add one more line as this suggestion, we end up with 2 empty lines.

Comment (jonahwilliams) [reply packages/flutter_tools/templates/plugin/test/projectName_test.dart.tmpl]: My mistake!

File: packages/flutter_tools/templates/plugin/lib/projectName.dart.tmpl (hunk 4 above)
Comment (truongsinh) [reply packages/flutter_tools/templates/plugin/lib/projectName.dart.tmpl]: resolve in latest commit

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Comment by zanderso** [comment]
@truongsinh @jonahwilliams Can this be landed independently of https://github.com/flutter/flutter/pull/26487 and https://github.com/flutter/flutter/pull/26466?

**Comment by jonahwilliams** [comment]
Good to land

**Pull request merged**